environment/src/main/java/com/sequenceiq/environment/environment/encryption/EnvironmentEncryptionService.java

```diff
@@ -92,6 +92,11 @@ DiskEncryptionSetCreationRequest createEncryptionResourcesCreationRequest(Enviro
         } else {
             builder.withEncryptionKeyResourceGroupName(diskEncryptionSetResourceGroupName);
         }
+        if (diskEncryptionSetResourceGroupName != null) {
+            builder.withIsSingleResourceGroup(Boolean.TRUE);
+        } else {
+            builder.withIsSingleResourceGroup(Boolean.FALSE);
+        }
         return builder.build();
     }
 
```

Comment:
As per my above comments, a separate `CloudResource` of type `ResourceType.AZURE_RESOURCE_GROUP` should be persisted for the DES RG. During deletion, this additional `CloudResource` needs to be looked up in `getResourcesForDeletion()` and passed over inside `DiskEncryptionSetDeletionRequest` (in addition to the `AZURE_DISK_ENCRYPTION_SET` resource).